driver/convert.c

```diff
@@ -1582,21 +1578,19 @@ static SQLRETURN parse_iso8601_timestamp(esodbc_stmt_st *stmt, xstr_st *xstr,
 	}
 
 	if (! to_utc) {
-		/* apply local offset */
-		tm.tm_sec -= _tz_dst_offt; /* serialize_statement() will update it */
-		tm.tm_isdst = -1; /* let the system determine the daylight saving */
-		/* rebalance member values */
-		if (mktime(&tm) == (time_t)-1) {
-			ERRH(stmt, "failed to adjust timestamp `" LCPDL "` by TZ (%ld). "
-				MKTIME_FAIL_MSG, LCSTR(&ts_str), _tz_dst_offt);
-			RET_HDIAG(stmt, SQL_STATE_22008, "Timestamp timezone adjustment "
-				"failed. " MKTIME_FAIL_MSG, 0);
+		/* convert UTC to localtime */
+		if ((utc = timegm(&tm)) == (time_t)-1 || ! (tmp = localtime(&utc))) {
```

Comment:
On Windows `localtime()` returns a pointer to a thread local static buffer.  On Linux it returns a pointer to a _global_ static buffer, so is not thread safe, and `localtime_r()` needs to be used instead.  This is not an immediate problem as you are only supporting Windows, but I think it would be worth adding a `TODO` to say it needs changing if Linux is ever supported.

(To complicate matters, Windows and C11 also have `localtime_s()`, but with the arguments the opposite way around.  Argh!)

Reply:
Thanks, Dave.
Very good points. I will actually add compile errors (for `! _WIN32`), since the TODO might be easy to oversee and the resulting issues then harder to identify; but in the next PR, since I moved that code a bit around and would keep the diff simple.